docs/contributing/Android-API-And-Related-Versions.md

```diff
@@ -120,3 +120,46 @@ ext.kotlin_version = "1.7.10"
 ...
 classpath "org.jetbrains.kotlin:kotlin-gradle-plugin:$kotlin_version"
 ```
+
+### compileOptions and kotlinOptions
+
+- sourceCompatibility must use JavaVersion.*
+- targetCompatibility must use JavaVersion.*
+- kotlinOptions jvmTarget should match the versions used by compileOptions or there should be a comment explaining why.
+- jvmTarget should use JavaVersion.<SOMEVERSION>.toString() or there should be a comment explaining why.
```

Comment:
It may be useful to include examples as to why they would be different if there's a reason besides things getting out of date.

Also, I know you're not fixing the `kotlinOptions jvmTarget` usages in this PR but what do you think about leaving comments or todos on those usages as this documentation suggests should be done?

Reply:
If I had examples I would include them but for now I am allowing for me to not know every situation and building in a "break glass" if there is a good reason. 

If I was going to modify each file I would just change the implementation rather than leave todos. 
That work is done in https://github.com/flutter/flutter/pull/164200

Reply:
Fair enough! Was wondering if there was a blocker.